src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -404,16 +405,17 @@ APPLICATION_INFO::FindNativeAssemblyFromHostfxr(
             m_pConfiguration->QueryHostFxrArgCount(),
             m_pConfiguration->QueryHostFxrArguments(),
             struNativeSearchPaths.QueryStr(),
-            dwBufferSize
+            dwBufferSize,
+            &dwRequiredBufferSize
         );
 
         if (intHostFxrExitCode == 0)
         {
             break;
         }
-        else if (intHostFxrExitCode == API_BUFFER_TOO_SMALL)
+        else if (dwRequiredBufferSize >= dwBufferSize)
```

Comment:
why >=? is the null terminator is not included?

Reply:
fixed.

Reply:
if the buffer copy is just char copy not a string, you need to add the null terminator otherwise you may run into buffer overrunning

Reply:
It is null terminated.